plume-models/src/posts.rs

```diff
@@ -864,113 +839,111 @@ impl Post {
+
+    fn from_activity((conn, searcher): &(&'a Connection, &'a Searcher), article: LicensedArticle, _actor: Id) -> Result<Post> {
         let license = article.custom_props.license_string().unwrap_or_default();
         let article = article.object;
-        if let Some(post) = Post::find_by_ap_url(
+        if let Ok(post) = Post::find_by_ap_url(
             conn,
             &article.object_props.id_string().unwrap_or_default(),
         ) {
-            post
+            Ok(post)
         } else {
             let (blog, authors) = article
                 .object_props
-                .attributed_to_link_vec::<Id>()
-                .expect("Post::from_activity: attributedTo error")
+                .attributed_to_link_vec::<Id>()?
                 .into_iter()
                 .fold((None, vec![]), |(blog, mut authors), link| {
                     let url: String = link.into();
                     match User::from_url(conn, &url) {
-                        Some(user) => {
-                            authors.push(user);
+                        Ok(u) => {
+                            authors.push(u);
                             (blog, authors)
-                        }
-                        None => (blog.or_else(|| Blog::from_url(conn, &url)), authors),
+                        },
+                        Err(_) => {
+                            (blog.or_else(|| Blog::from_url(conn, &url).ok()), authors)
+                        },
                     }
                 });
 
             let cover = article
                 .object_props
                 .icon_object::<Image>()
                 .ok()
-                .and_then(|img| Media::from_activity(conn, &img).map(|m| m.id));
+                .and_then(|img| Media::from_activity(conn, &img).ok().map(|m| m.id));
 
             let title = article
                 .object_props
-                .name_string()
-                .expect("Post::from_activity: title error");
+                .name_string()?;
             let post = Post::insert(
                 conn,
                 NewPost {
-                    blog_id: blog.expect("Post::from_activity: blog not found error").id,
+                    blog_id: blog?.id,
                     slug: title.to_kebab_case(),
                     title,
                     content: SafeString::new(
                         &article
                             .object_props
-                            .content_string()
-                            .expect("Post::from_activity: content error"),
+                            .content_string()?,
                     ),
                     published: true,
                     license: license,
                     // FIXME: This is wrong: with this logic, we may use the display URL as the AP ID. We need two different fields
-                    ap_url: article.object_props.url_string().unwrap_or_else(|_|
+                    ap_url: article.object_props.url_string().unwrap_or(
```

Comment:
I don't know if there are other places where `unwrap_or_else(|_| ..)` was replaced with `unwrap_or(..)`, if so this comment also apply to them. I think it would be better to `or_else(|_| ..)?`, I think you already did it at other places